.github/workflows/commitlint.yml

```diff
@@ -8,24 +7,4 @@ on:
 
 jobs:
   commitlint:
-    runs-on: ubuntu-20.04
-    steps:
-      - name: Check out the repo
-        uses: actions/checkout@v3
-        with:
-          # Fetch 100 commits.  Should be enough?
-          fetch-depth: 100
-
-      # This step is needed to fix https://github.com/conventional-changelog/commitlint/issues/3256
-      - name: Remove tsconfig file
-        run: rm tsconfig.json
```

Comment:
https://github.com/openedx/.github/blob/master/.github/workflows/commitlint.yml does not include this step.

The issue mentioned here (https://github.com/conventional-changelog/commitlint/issues/3256) is still open, but I tested the repro steps (`commitlint --from HEAD~2 --to HEAD`) locally and found no errors.

The environment mentioned in the linked issue was:
| | |
| - | - |
| `commitlint --version` | 16.3.0 |
| `git --version` | 2.36.1 |
| `node --version` | 18.4.0 |

After running `nvm use`, my local environment was
| | |
| - | - |
| `commitlint --version` | 18.4.3 |
| `git --version` | 2.42.0 |
| `node --version` | 18.18.2 |

The commitlint gh action  [uses commitlint version `^18.2.0`](https://github.com/wagoid/commitlint-github-action/blob/0d749a1a91d4770e983a7b8f83d4a3f0e7e0874e/package.json#L46), and [node version `20.9.0`](https://github.com/wagoid/commitlint-github-action/blob/0d749a1a91d4770e983a7b8f83d4a3f0e7e0874e/.github/workflows/commitlint.yml#L13), so overall this feels safe to remove.